### PR TITLE
Modify notations

### PR DESCRIPTION
**Pull request by vltanh**
I modified some of the notations so that it will be less confusing (I was confused reading it).

#### What does this implement/fix?

- It is better to specify of which the Jacobian matrix is, here it is the transition and measurement function `f` and `g` (writing it as `J_F` and `J_H` is misleading)
- It is better to use `Delta` so as not to confuse people with the `d` commonly used in derivative
- It is more correct to use `\partial` instead of `d` since it is more multivariate than it is not
- Fix minor errors like extra brackets

### CODE REVIEW

File: docs/modules/extended_kalman_filter_localization.rst [hunk 1]
```diff
@@ -77,32 +77,36 @@ Motion Model
...
 
-:math:`dt` is a time interval.
+:math:`\Delta t` is a time interval.
 
 This is implemented at
 `code <https://github.com/AtsushiSakai/PythonRobotics/blob/916b4382de090de29f54538b356cef1c811aacce/Localization/extended_kalman_filter/extended_kalman_filter.py#L53-L67>`__
 
+The motion function is that
+
+:math: `\begin{equation*} \begin{bmatrix} x' \\ y' \\ w' \\ v' \end{bmatrix} = f(\textbf{x}, \textbf{u}) = \begin{bmatrix} x + v\cos(\phi)\Delta t \\ y + v\sin(\phi) \\ \phi + \omega \Delta t \\ v \end{bmatrix} \end{equation*}`
```
Comment (AtsushiSakai) [inline docs/modules/extended_kalman_filter_localization.rst:103]: Maybe this space is not needed.
```suggestion
:math:`\begin{equation*} \begin{bmatrix} x' \\ y' \\ w' \\ v' \end{bmatrix} = f(\textbf{x}, \textbf{u}) = \begin{bmatrix} x + v\cos(\phi)\Delta t \\ y + v\sin(\phi) \\ \phi + \omega \Delta t \\ v \end{bmatrix} \end{equation*}`
```

File: docs/modules/extended_kalman_filter_localization.rst [hunk 2]
```diff
@@ -111,15 +115,19 @@ The robot can get x-y position infomation from GPS.
...
+.. math:: \textbf{z}_{t} = g(\textbf{x}_t) = H \textbf{x}_t
 
 where
 
-:math:`\begin{equation*} B= \begin{bmatrix} 1 & 0 & 0& 0\\ 0 & 1 & 0& 0\\ \end{bmatrix} \end{equation*}`
+:math:`\begin{equation*} H = \begin{bmatrix} 1 & 0 & 0 & 0 \\ 0 & 1 & 0 & 0 \\ \end{bmatrix} \end{equation*}`
+
+The observation function states that
+
+:math: `\begin{equation*} \begin{bmatrix} x' \\ y' \end{bmatrix} = g(\textbf{x}) = \begin{bmatrix} x \\ y \end{bmatrix} \end{equation*}`
```
Comment (AtsushiSakai) [inline docs/modules/extended_kalman_filter_localization.rst:126]: Maybe this space is not needed.
```suggestion
:math:`\begin{equation*} \begin{bmatrix} x' \\ y' \end{bmatrix} = g(\textbf{x}) = \begin{bmatrix} x \\ y \end{bmatrix} \end{equation*}`
```

Comment (AtsushiSakai) [reply docs/modules/extended_kalman_filter_localization.rst:126]: In my environment, this space might cause rendering issue:
SS 2021-10-03 at 16 49 13

**Review by AtsushiSakai — CHANGES_REQUESTED** [review]
Thank you for your PR. I agree your notations look better. I have some commnets. PTAL.

**Comment by vltanh** [comment]
I agree, it was because I am not familiar with this format so I thought it does not matter. Thank you for pointing that out.

**Review by AtsushiSakai — APPROVED** [review]
LGTM. Thanks.

**Pull request merged**